src/state/slices/opportunitiesSlice/selectors/stakingSelectors.ts

```diff
@@ -619,9 +619,9 @@ export const selectAggregatedEarnUserStakingEligibleOpportunities = createDeepEq
     const eligibleOpportunities = aggregatedEarnUserStakingOpportunities.reduce<
       StakingEarnOpportunityType[]
     >((acc, opportunity) => {
-      const hasBalance = opportunity.underlyingAssetIds.some(
-        assetId => !bnOrZero(assetBalances[assetId]).isZero(),
-      )
+      const hasBalance = opportunity.underlyingAssetIds
+        .concat(opportunity.underlyingAssetId)
+        .some(assetId => !bnOrZero(assetBalances[assetId]).isZero())
```

Comment:
@NeOMakinG since ROON is now the odd one, we may want to add a more generalized notion of `stakingAssetIds[]` on top of the existing `assetId` / `underlyingAssetId` / `underlyingAssetIds[]`, WDYT?

The only reason why the quote-on-quote "abstraction" works atm is because each DeFi protocol has its own logic, where it defines internally *what* is/are the staking AssetId/AssetIds (i.e `underlyingAssetId` or `underlyingAssetIds[]`.

We've discussed making the abstraction an actual abstraction many many times, and that would be a nice baby step in this direction, could potentially kill a bit of code while at it.

Reply:
That could be cool yes, this is a real baby step btw 😆